### PR TITLE
[DO NOT MERGE] Feature/empty branch live test

### PR DESCRIPTION
**Pull request by martinda**
An attempt at writing a live test for the empty branch scenario.
CC @cdancy

### CODE REVIEW

File: src/test/java/com/cdancy/bitbucket/rest/TestUtilities.java [hunk 1]
```diff
@@ -161,14 +163,47 @@ public static synchronized GeneratedTestContents initGeneratedTestContents(final
...
+                                                                                final BitbucketApi api) {
+        assertThat(endpoint).isNotNull();
+        assertThat(credential).isNotNull();
+        assertThat(api).isNotNull();
+
+        // Use the user provided test project, or create one
+        GeneratedTestProject generatedTestProject = maybeCreateTestProject(endpoint, credential, api);
+
+        // create test repo
+        Repository testRepository = createTestRepository(api, generatedTestProject.project);
```
Comment (martinda) [inline src/test/java/com/cdancy/bitbucket/rest/TestUtilities.java:204]: @cdancy the code would let the test call `createRepository` without adding content, thus it would be suitable for testing the empty branch scenario.

**Review by martinda — COMMENTED** [review]
@cdancy What do you think?

**Comment by martinda** [comment]
Closing this one in favor of https://github.com/cdancy/bitbucket-rest/pull/148
